error_code_report-0.1.0.json

```
@@ -0,0 +1,92 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "https://schemas.exasol.com/error_code_report-0.1.0.json",
+  "type": "object",
+  "title": "Exasol error code report schema.",
+  "description": "This schema defines the structure of a report about error codes in an Exasol project.",
+  "additionalProperties": false,
+  "required": [
+    "projectName",
+    "projectVersion",
+    "errorsCodes"
+  ],
+  "properties": {
+    "$schema": {
+      "type": "string",
+      "description": "Reference to the schema. Setting this helps your IDE to assist you while writing the definition.",
+      "enum": [
+        "https://schemas.exasol.com/error_code_report-0.1.0.json",
+        "./error_code_report-0.1.0.json"
+      ]
+    },
+    "projectName": {
+      "type": "string",
+      "title": "Name of the project.",
+      "description": "Typically the GitHub repository name."
+    },
+    "projectVersion": {
+      "type": "string",
+      "title": "Version tag of the reported release."
+    },
+    "errorsCodes": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/errorCode"
+      },
+      "description": "List of `errorCodes` declared in this project version."
+    }
+  },
+  "definitions": {
+    "errorCode": {
+      "type": "object",
+      "properties": {
+        "errorCode": {
+          "type": "string",
+          "pattern": "^[WFE]-[A-Z0-9]+(-[A-Z0-9]+)?-\\d$"
```

```suggestion
          "pattern": "^[WFE]-[A-Z][A-Z0-9]*(-[A-Z][A-Z0-9]*)?-\\d$"
```
This small change makes sure you get Identifier-quality short tag fragments acceptable in all programming languages. Might come in handy if you ever need some kind of relation between the two.

Or identifier-quality in databases or search engines maybe.

Think of it as a precaution.

---

@redcatbear But would you need for identifier quality not something else than the dash as delimiter? Or, do you mean in general, that we could replace the dashes if needed for the language, but keep the tags as is?

---

What I was thinking about is if you needed to use the individual pieces as identifiers, they should not start with numbers, since most parsers choke on that.
In `E-FOO-BAR1-400` for example all short tags can be Identifiers in all languages.
If you chose `E-FOO-1BAR-400`, the `1BAR` part can be a problem.
I am not sure whether I am overly careful here. It's either instinct after years of experience or -- well just paranoia. :-)